### PR TITLE
add test config to alloy smithy model as metadata

### PR DESCRIPTION
**Pull request by yisraelU**
Instead of storing a separate file in the published jar , add the data to the metadata of the published smithy model

### CODE REVIEW

File: modules/protocol-tests/resources/META-INF/smithy/manifest [hunk 1]
```diff
@@ -7,4 +7,5 @@ HeaderEndpoint.smithy
 Health.smithy
 Pizza.smithy
 RoundTrip.smithy
-Version.smithy
\ No newline at end of file
+Version.smithy
+test-config.json
```
Comment (daddykotex) [inline modules/protocol-tests/resources/META-INF/smithy/manifest:11]: How does that work? The file does not exist anymore but we keep a reference to it in the manifest.

Is it possible you moved the file and forgot to include the new one in the commit?

Comment (yisraelU) [reply modules/protocol-tests/resources/META-INF/smithy/manifest:11]: just bad commit

**Review by daddykotex — APPROVED** [review]

File: modules/protocol-tests/resources/META-INF/smithy/test-config.json [hunk 2]
```diff
@@ -0,0 +1,1227 @@
+{
+  "smithy": "2.0",
+  "metadata": {
+    "restJsonAllowList": [
```
Comment (Baccata) [inline modules/protocol-tests/resources/META-INF/smithy/test-config.json:4]: we should adopt the practice of prefixing the metadata in this repo with "alloy"

Comment (yisraelU) [reply modules/protocol-tests/resources/META-INF/smithy/test-config.json:4]: good point

**Pull request merged**